language/defined_spec.rb

```diff
@@ -900,6 +900,14 @@
     defined?(DefinedSpecs::Undefined).should be_nil
   end
 
+  it "returns nil when the constant is not defined and the parent implements const_missing" do
+    def DefinedSpecs.const_missing(const)
+      const
+    end
+
+    defined?(DefinedSpecs::Undefined).should be_nil
+  end
```

Comment:
Dynamic defining a new method in a class/module shared by multiple test cases may make them surprisingly coupled, what makes sense to avoid.

So I would add a new fixture module/class specifically for this test case, e.g. `DefinedSpecs::ModuleWithConsMissing` instead.